pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -275,12 +279,41 @@ func computeActiveCondition(pa *pav1alpha1.PodAutoscaler, pc podCounts) {
 }
 
 // activeThreshold returns the scale required for the pa to be marked Active
-func activeThreshold(pa *pav1alpha1.PodAutoscaler) int {
+func (c *Reconciler) activeThreshold(ctx context.Context, pa *pav1alpha1.PodAutoscaler) int {
+	logger := logging.FromContext(ctx)
+	initScale := int(config.FromContext(ctx).Autoscaler.InitialScale)
+	allowZero := config.FromContext(ctx).Autoscaler.AllowZeroInitialScale
+	revInitScale, ok := pa.ObjectMeta.Annotations[autoscaling.InitialScaleAnnotationKey]
+	defaultScale := 1
+	if ok {
+		initialScaleInt, err := strconv.Atoi(revInitScale)
+		if err != nil {
+			logger.Errorf("error processing initialScale annotation: %v", err)
+		}
+		if initialScaleInt == 0 {
+			if allowZero {
+				initScale = 0
+			}
+		} else if initialScaleInt > 1 {
+			initScale = initialScaleInt
+		}
+	}
+
+	if c.scaler.lastReconcileTime != nil {
+		c.scaler.initialScaleTime += time.Since(*c.scaler.lastReconcileTime)
+		if c.scaler.initialScaleTime < config.FromContext(ctx).Autoscaler.StableWindow {
+			defaultScale = initScale
+		}
+	} else {
```

Comment:
scaler is a global variable. It's not per-revision, I don't think you can use it that way.